WebService/Sources/WebService/Components/AccountHandlers/CreateAccount.swift

```diff
@@ -17,18 +18,18 @@ struct CreateAccount: Handler {
     
     
     @Environment(\.xpenseModel) var xpenseModel
-    @Environment(\.connection) var connection
     
     @Parameter(.http(.body)) var account: CreateAccountMediator
     
-    @Throws(.unauthenticated, reason: "The User is not Authenticated correctly") var userNotFound: ApodiniError
-    
+    var user = Authorized<User>()
```

Comment:
One thought I had while looking at the code: Maybe it would be cool to make `Authorized` a property wrapper? I am not quite sure how this works under the hood but in line with all the other property wrappers that we have an `@Authorized var user: User` would be very much in line with all the features we have in Apodini.

Reply:
I agree with you, that a property wrapper based solution would be ideal, and doing a `@Authorized var user: User` was actually my first plan and goal, as its the most straightforward solution with not much complexity in the syntax.

However this approach, as of right now, is blocked by the fact that [[SE-0310] Effectful Readonly Properties #ExtensionsConsidered](https://github.com/kavon/swift-evolution/blob/main/proposals/0310-effectful-readonly-properties.md#extensions-considered) doesn't include support for the `wrappedValue` of propertyWrappers, meaning I can't annotate the computed `wrappedValue` with a `throws` effect. It is not clear if or when such a feature would land.

A authorized property, independent of its implementation, must provide the following two functionalities:
* Accessing the Authenticatable instance might throw. A Handler might request an authorized instance which is not present. This can happen simply when a `Authorize` Metadata was forgotten, or if a `AuthorizeOptionally` was used but the Handler nonetheless enforces presence of an authorized Authenticatable by requesting it.  
Currently this is implemented by using a throwing `callAsFunction` on the Authorized property (as seen in the PR).
* The property must provide access to authorization attributes. Currently the only one present is the `isAuthorized` property. It can be used e.g. in concoction with an `AuthorizeOptionally` Metadata to explicitly check if an authorized instance is available and therefore make certain functionality of the Handler only available to authorized requests.

If a propertyWrapper based approach is preferred in any case one could think of returning some sort of wrapper structure in the `wrappedValue`. Using a declaration like the following:
```swift
@Authorized(User.self) var user
```
The type of the `wrappedValue` could be something like a `AuthorizedAuthenticatable<User>` which exposes similar functionality like the current Authorized property: throwing `callAsFunction` to access the instance and a `isAuthorized` property.

What do you think?

Reply:
Thanks for the detailed answer! I have not considered that the access to the user property is done using `callAsFunction` and therefore can throw. 
I have come across this multiple times: It would be very great of property wrappers could also have wrapped values that can be `async` or `throw` in case of an error. This would e.g. also nicely enable us to lazily decode `@Parameter` properties. 

I think that making type of the `wrappedValue` something like a `AuthorizedAuthenticatable<User>` sounds like the best solution to be but it doesn't come close to the possibility to `throw` while accessing the `wrappedValue`. 
I would be fine with that but on the other hand think that this might just be future work once property wrappers support this feature and we should note this down as something to tackle in the future. The same probably also applies to the `Delegate` feature.

Reply:
I created a pull request in Apodini to incorporate the discussed changes: https://github.com/Apodini/Apodini/pull/332.
I would request a quick review from you once the CI completed successfully.

I plan to link to this PR (or the resulting commit) from my thesis from e.g. a footnote such that I can present the complete change set required to migrate ApodiniAuthorization to an interested reader. Hence the aim to match the "final" syntax in this PR.

Reply:
Sounds good, thanks a lot!

Reply:
https://github.com/Apodini/Apodini/pull/332 has landed. Will update the PR to incorporate those changes and merge afterwards.